types/src/leb128.rs

```diff
@@ -0,0 +1,99 @@
+use crate::error::TypesError;
+use crate::error::TypesError::NotEnoughData;
+use bytes::{Buf, BufMut};
+
+#[inline]
+pub fn read_leb128(buffer: &mut &[u8]) -> Result<u64, TypesError> {
+    let mut value = 0u64;
+    let mut shift = 0;
+    loop {
+        if buffer.remaining() < 1 {
+            return Err(NotEnoughData(
+                "decoding LEB128, 0 bytes remaining".to_string(),
+            ));
+        }
+        let byte = buffer.get_u8();
+        value |= (byte as u64 & 0x7f) << shift;
+        if byte & 0x80 == 0 {
+            break;
+        }
+        shift += 7;
+        if shift > 57 {
+            return Err(NotEnoughData("decoding LEB128, invalid shift".to_string()));
```

Comment:
I don't think that the shift value is invalid here. It's unsupported by the crate (and totally unrealistic), but I don't see any guarantees in the CH documentation